sdk/python/feast/client.py

```diff
@@ -400,27 +428,105 @@ def get_batch_features(
         return Job(response.job, self._serving_service_stub)
 
     def _validate_entity_rows_for_batch_retrieval(
-        self, entity_rows, feature_sets_request
+        self, entity_rows: pd.DataFrame, feature_sets_request
     ):
         """
-        Validate whether an entity_row dataframe contains the correct
-        information for batch retrieval
+        Validate whether an entity_row DataFrame contains the correct
+        information for batch retrieval.
+
+        Datetime column must be present in the DataFrame.
 
         Args:
-            entity_rows: Pandas dataframe containing entities and datetime
-                column. Each entity in a feature set must be present as a
-                column in this dataframe.
-            feature_sets_request: Feature sets that will be requested
+            entity_rows (pd.DataFrame):
+                Pandas DataFrame containing entities and datetime column. Each
+                entity in a feature set must be present as a column in this
+                DataFrame.
+
+            feature_sets_request:
+                Feature sets that will be requested.
         """
 
+        self._validate_columns(
+            columns=entity_rows.columns,
+            feature_sets_request=feature_sets_request,
+            datetime_field="datetime"
+        )
+
+    def _validate_avro_for_batch_retrieval(
+            self, source: str, feature_sets_request
```

Comment:
This method name is inconsistent with the one above which mentions entity_rows

Reply:
Renamed `_validate_entity_rows_for_batch_retrieval` to `_validate_dataframe_for_batch_retrieval`.

As such, this should be consistent in what the function is doing, which is to ensure that right columns are present in different file types.